File: php/iforevents/src/IntegrationResult.php
<?php

declare(strict_types=1);

namespace IForevents;

/** Outcome of one integration call; the facade never throws for these. */
final class IntegrationResult
{
    public readonly \DateTimeImmutable $timestamp;

    public function __construct(public readonly string $integration, public readonly bool $success, public readonly ?\Throwable $error = null)
    {
        $this->timestamp = new \DateTimeImmutable('now', new \DateTimeZone('UTC'));
    }
}

File: java/iforevents/src/main/java/com/iforevents/IntegrationResult.java
package com.iforevents;

import java.util.Date;

/** Outcome of one integration call; the facade never throws for these. */
public final class IntegrationResult {
    public final String integration;
    public final boolean success;
    public final Throwable error;
    public final Date timestamp = new Date();

    public IntegrationResult(String integration, boolean success, Throwable error) {
        this.integration = integration;
        this.success = success;
        this.error = error;
    }

    @Override
    public String toString() {
        return "IntegrationResult{" + integration + ", success=" + success + (error == null ? "" : ", error=" + error) + "}";
    }
}
